### ISSUE TITLE
Doesn't work with named outlet compoent

### ISSUE THREAD

**Issue by musaffa** [salsify/ember-css-modules#82]
Named outlet is used to render specific templates in specific routes. For example:
```Haml
// application/template.em
= outlet 'nav-bar'
= outlet
```
For all the routes a component is set for the `nav-bar` outlet in the application route:
```Js
// application/route.js
renderTemplate() {
  this._super(...arguments);
  this.render('components/navs/nav-bar', {
    into: 'application',
    outlet: 'nav-links'
  });
}
```
In the admin route a different component is set for the `nav-bar` outlet:
```Js
// admin/route.js
renderTemplate() {
  this._super(...arguments);
  this.render('components/admin/navs/nav-bar', {
    into: 'application',
    outlet: 'nav-links'
  });
}
```

Thus the navigation bar in the admin panel is rendered differently to the rest of the application.

`ember-css-modules` doesn't currently work in these outlet components.

**Comment by dfreeman** [comment]
Hi @musaffa! Named outlets actually use controllers rather than components to power their templates (which is why you're having to include that `components/` part in the path), but I would expect them to still work with ember-css-modules.

What paths do your `nav-bar` template and styles live at?

**Comment by musaffa** [comment]
```
└── nav-bar
    ├── component.js
    ├── styles.css
    └── template.em
```

**Comment by dfreeman** [comment]
As I mentioned, that `component.js` isn't likely to be doing anything, since `render()` is going to look for a controller instead.

If it doesn't find one, I believe it will default to using the `into:` controller to power your template instead. Since the local classes that are applied depend on the component or controller that's supplying data to a template, you could do one of two things:

 - add a `controller.js` into your `nav-bar` directory (even as simple as just `export default Ember.Controller`)
 - merge your `styles.css` that you're using for `nav-bar` into the one you're using for your `admin` route/controller

Alternatively, you could use something like `ember-wormhole` instead of `render()` to make your `nav-bar` an actual component.

**Comment by musaffa** [comment]
Making it a controller actually works. But I want to keep it as a component. Don't you think that using `ember-wormhole` for `nav-bar` a bit ugly hack? Or is it perfectly normal to use it to do such thing?

**Comment by dfreeman** [comment]
I think of it as a pretty normal use case for ember-wormhole. I like that it means I only have to look one place (the template) to determine what a route is going to render, and it also prevents confusion from `render()` like this issue, where it's not obvious what context a template is actually rendering in.

But of course it's ultimately your call—using `render()` with named outlets seems to have fallen out of vogue, but as far as I know it's not deprecated or anything, so to each their own 🙂

**Comment by musaffa** [comment]
It looks like this.render is going to be [deprecated](https://emberjs.com/deprecations/v2.x/#toc_rendering-into-a-render-helper-that-resolves-to-an-outlet) in Ember 3.0.

**Comment by dfreeman** [comment]
If I read the blog post right, the only thing that's actually deprecated is using `{{render 'name'}}` instead of `{{outlet 'name'}}` in your template. That said, they do call out that named outlets aren't really considered a best practice any more:

> In general, the pattern of named outlets or named render helpers is discouraged.

Given that, I'd lean toward `ember-wormhole` or `ember-elsewhere` for your use case.
